src/test/rbd_mirror/image_replayer/journal/test_mock_PrepareReplayRequest.cc

```diff
@@ -252,13 +287,35 @@ TEST_F(TestMockImageReplayerJournalPrepareReplayRequest, ResyncRequested) {
   MockStateBuilder mock_state_builder(mock_local_image_ctx,
                                       mock_remote_journaler,
                                       mirror_peer_client_meta);
+
+  // check initial state
+  expect_is_resync_requested(mock_journal, true, 0);
+  // remote is not primary, so no resync should occur
+  expect_is_remote_primary(mock_state_builder, false);
+  expect_journal_get_tag_tid(mock_journal, 345);
+  expect_journal_get_tag_data(mock_journal, {"remote mirror uuid"});
+
+  librbd::journal::ClientData client_data{
```

Comment:
Nit: preserve "lookup remote image tag class" and "single promotion event" (?) comments from where this snippet is copied ;)